build-tools/packages/build-cli/src/commands/updates/packageTypes.ts

```diff
@@ -0,0 +1,157 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Flags } from "@oclif/core";
+import { Package } from "@fluidframework/build-tools";
+import path from "node:path";
+import { PackageCommand } from "../../BasePackageCommand";
+import * as fs from "fs";
+
+/**
+ * A enum defining the four release types
+ */
+enum UpdatePackageJsonEnum {
+	ALPHA = "alpha",
+	BETA = "beta",
+	DEFAULT = "public",
+}
+
+/**
+ * A enum defining the api-extractor.json configs
+ */
+enum ApiExtractorConfig {
+	DTSROLLUP = "dtsRollup",
+	ALPHATRIMMED = "alphaTrimmedFilePath",
+	BETATRIMMED = "betaTrimmedFilePath",
+}
+
+/**
+ * A typeguard to check if a string is a {@link UpdatePackageJsonEnum}.
+ */
+function isUpdatePackageJsonTypes(type: string): type is UpdatePackageJsonEnum {
+	return (
+		type === UpdatePackageJsonEnum.ALPHA ||
+		type === UpdatePackageJsonEnum.BETA ||
+		type === UpdatePackageJsonEnum.DEFAULT
+	);
+}
+
+export default class UpdatePackageTypesCommand extends PackageCommand<
+	typeof UpdatePackageTypesCommand
+> {
+	static readonly description = "Updates the types in package.json based on the release flag";
+
+	static flags = {
+		path: Flags.directory({
+			description: "Path to a directory containing a package.",
+			exists: true,
+			required: true,
+		}),
+		types: Flags.custom<UpdatePackageJsonEnum>({
+			description:
+				"The types flag is used to specify the type of release when updating a package.json file. It accepts a custom enumeration called UpdatePackageJsonEnum that defines different release types.",
+			parse: async (input) => {
+				if (isUpdatePackageJsonTypes(input)) {
+					return input;
+				}
+				throw new Error(`Invalid release type: ${input}`);
+			},
+		})(),
+		...PackageCommand.flags,
+	};
+
+	private readonly packageTypesUpdated: string[] = [];
+
+	protected async processPackage(pkg: Package): Promise<void> {
+		// Define paths to package.json and api-extractor.json
+		const packageJsonFilePath = path.join(pkg.directory, "package.json");
+		const apiExtractorConfigFilePath = path.join(pkg.directory, "api-extractor.json");
+
+		// Check if api-extractor.json exists in the package directory
+		const apiExtractorConfigExists = fs.existsSync(apiExtractorConfigFilePath);
+
+		try {
+			// Read the content of the package.json file
+			const packageJsonContent = fs.readFileSync(packageJsonFilePath, "utf-8");
+
+			// Parse the JSON content to an object
+			const packageJson = JSON.parse(packageJsonContent);
+
+			if (apiExtractorConfigExists) {
+				// Read the content of api-extractor.json
+				const apiExtractorConfigContent = fs.readFileSync(
```

Comment:
Maybe there is a type for this in the api-extractor package that we could use for stronger typing? If not you should make a minimal interface and use it. I speak from experience that `any` _will_ mess with your day given the chance. :smile:

Reply:
casting this to string 😄